src/rules/use-client.ts

```diff
@@ -113,12 +116,97 @@ const create = Components.detect(
       });
     }
 
+    function findFirstParentOfType(
+      node: Rule.Node,
+      type: string
+    ): Rule.Node | null {
+      let currentNode: Rule.Node | null = node;
+
+      while (currentNode) {
+        if (currentNode.type === type) {
+          return currentNode;
+        }
+        currentNode = currentNode?.parent;
+      }
+
+      return null;
+    }
+
+    function isNodeInTree(node: Rule.Node, target: Rule.Node): boolean {
+      let currentNode: Rule.Node | null = node;
+
+      while (currentNode) {
+        if (currentNode === target) {
+          return true;
+        }
+        currentNode = currentNode.parent;
+      }
+
+      return false;
+    }
+
+    function getBinaryBranchExecutedOnServer(node: BinaryExpression): {
+      isGlobalClientPropertyCheck: boolean;
+      serverBranch: Rule.Node | null;
+    } {
+      const isGlobalClientPropertyCheck =
+        node.left?.type === "UnaryExpression" &&
+        node.left.operator === "typeof" &&
+        node.left.argument?.type === "Identifier" &&
+        validGlobalsForServerChecks.has(node.left.argument?.name) &&
+        node.right?.type === "Literal" &&
+        node.right.value === "undefined" &&
+        (node.operator === "===" || node.operator === "!==");
+
+      let serverBranch = null;
+
+      if (!isGlobalClientPropertyCheck) {
+        return { isGlobalClientPropertyCheck, serverBranch };
+      }
+
+      //@ts-expect-error
+      const { parent } = node;
+      if (!parent) {
```

Comment:
Does this fix the error?
```suggestion
      const parent = node?.parent
      if (!parent) {
```

Reply:
No, the error comes from the `parent` field apparently not being a part of the `BinaryExpression` type 🤔 

`Property 'parent' does not exist on type 'BinaryExpression'.`